api/src/models/member.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Schema representing a H4I member
+ */
+const mongoose = require('mongoose');
+
+const levelEnum = {
+  ADMIN: 'ADMIN',
+  DIRECTOR: 'DIRECTOR',
+  LEAD: 'LEAD',
+  MEMBER: 'MEMBER',
+  TBD: 'TBD',
+};
+
+const locationEnum = {
+  REMOTE: 'REMOTE',
+  CAMPUS: 'CAMPUS',
+  TBD: 'TBD',
+};
+
+const roleEnum = {
+  CO_DIRECTOR: 'CO_DIRECTOR',
+  COMMUNITY_DIRECTOR: 'COMMUNITY_DIRECTOR',
+  EXTERNAL_DIRECTOR: 'EXTERNAL_DIRECTOR',
+  ACADEMY_LEAD: 'ACADEMY_LEAD',
+  TECH_LEAD: 'TECH_LEAD',
+  PR_LEAD: 'PR_LEAD',
+  PRODUCT_MANAGER: 'PRODUCT_MANAGER',
+  PRODUCT_DESIGNER: 'PRODUCT_DESIGNER',
+  DEVELOPER: 'DEVELOPER',
+  ACADEMY_MEMBER: 'ACADEMY_MEMBER',
+};
+
+const statusEnum = {
+  ACTIVE: 'ACTIVE',
+  HIATUS: 'HIATUS',
+  RETIRED: 'RETIRED',
+  TBD: 'TBD',
+};
+
+const semesterEnum = {
+  FALL: 'FALL',
+  SPRING: 'SPRING',
+};
+
+const classStandingEnum = {
+  FRESHMAN: 'FRESHMAN',
+  SOPHOMORE: 'SOPHOMORE',
+  JUNIOR: 'JUNIOR',
+  SPRING_SENIOR: 'SPRING_SENIOR',
+  FALL_SENIOR: 'FALL_SENIOR',
+  MASTERS: 'MASTERS',
+  PHD: 'PHD',
+};
+
+const Member = new mongoose.Schema({
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  oauthID: { type: String, required: true, unique: true },
+  email: { type: String, required: true, unique: true },
+  phone: { type: String, required: true },
+  netID: { type: String, required: true },
+  UIN: { type: String, required: true },
+  major: { type: String, required: true },
+  birthdate: { type: Date, required: true },
+  github: { type: String, required: false },
+  snapchat: { type: String, required: false },
+  instagram: { type: String, required: false },
+  areDuesPaid: { type: Boolean, required: true },
+
+  gradYear: { type: Number, required: true },
+  gradSemester: {
+    type: String,
+    enum: [...semesterEnum],
```

Comment:
Just curious, is the spreading to ensure that the object gets copied?

Reply:
That was actually a mistake and would probably through an error, I was looking for something that would put the object values into an array, I changed it to Object.values instead.